### DISCUSSION TITLE
slice array in view collection

### DISCUSSION THREAD

**Discussion by antoineds1** [pocketbase/pocketbase#2838] [Q&A] [answered]
hi, I have a fitusers collection which contains the friends column which is a relationship table, I would like to create a view which contains the first 2-3 friends of each user, I succeeded with the command below to obtain a string table but not a relationship table:
`SELECT 
  id, profilPicture, username, bio, 
  json('["' || JSON_EXTRACT(friends, '$[0]') || '", "' || JSON_EXTRACT(friends, '$[1]') || '"]') as friends_array
FROM fitusers;`

id | k6pmk3hikgsific
-- | --
profilPicture | N/A
username | test
bio | N/A
friends_array | ["34xhao2jz7sa5t9","908cd7fnw43uy0j"]

thanks for the help

**Comment by ganigeorgiev** [comment]
Column expressions are always mapped as `json` field.

Eventually sometime in the future I may reconsider allowing user to explicitly define the view schema and its fields mapping, but currently this is not supported.

**Reply by antoineds1** [reply]
so I can't transform the currently string ids into relation ?

**Reply by ganigeorgiev** [reply, marked as answer]
No sorry, at the moment the view schema fields mappings cannot be changed (_I've added it to the roadmap and will consider it for v0.18/v0.19 release as it was requested by other users too_).
